tests/test_metadata/test_v2.py

```diff
@@ -316,3 +316,16 @@ def test_zstd_checksum() -> None:
         arr.metadata.to_buffer_dict(default_buffer_prototype())[".zarray"].to_bytes()
     )
     assert "checksum" not in metadata["compressor"]
+
+
+def test_structured_dtype_fill_value_serialization(tmp_path):
+    group_path = tmp_path / "test.zarr"
+    root_group = zarr.open_group(group_path, mode="w", zarr_format=2)
+    root_group.create_array(
+        name="structured_headers",
+        shape=(100, 100),
+        chunks=(100, 100),
+        dtype=np.dtype([("foo", "i4"), ("bar", "i4")]),
+    )
+
+    zarr.consolidate_metadata(root_group.store, zarr_format=2)
```

Comment:
this test is weak (it only tests that consolidate metadata doesn't error). Do you think it makes sense to make this test a big stronger, e.g. by checking that the fill value was actually encoded the way we think it should have been? 

Also I think we need a check to ensure that if the dtype is void and the fill value is `None`, then there's no base64 encoding (I know from the implementation in this PR that the test will pass, but it's good to have the test in any case)

Reply:
> this test is weak (it only tests that consolidate metadata doesn't error). Do you think it makes sense to make this test a big stronger, e.g. by checking that the fill value was actually encoded the way we think it should have been?

Oh wow, I totally intended to do that (hence the name of the test).

> Also I think we need a check to ensure that if the dtype is void and the fill value is None, then there's no base64 encoding

Great suggestion

Reply:
Thanks! tried both!